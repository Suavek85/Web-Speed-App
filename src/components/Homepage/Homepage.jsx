import React from 'react'
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'

import { tabletWidth } from'../../scss/mediaqueries'
import PrimaryCard from '../Primary/PrimaryCard'
import Side from '../Side/Side'
import Navigation from '../Navigation/Navigation'
import styles from './Homepage.scss'

export default function Homepage() {

  //STYLES
  const cx = classNames.bind(styles)
  const getButtonClasses = cx({
    block: true,
    blockTablet: useMediaQuery(tabletWidth),
  })

  return (
    <section className={ getButtonClasses }>
      <Navigation />
      <PrimaryCard />
      <Side />
    </section>
  )
}

