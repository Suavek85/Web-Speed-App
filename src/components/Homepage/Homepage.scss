@import '../../scss/styles';

$block: app-wrapper;
$blockTablet: app-wrapper--tablet;

.#{$block} {
  position: relative;
  display: flex;
  background-color: $color-main-background;

  &--tablet {
    flex-direction: column;
  }
}

:export {
  block: $block;
  blockTablet: $blockTablet;
}
