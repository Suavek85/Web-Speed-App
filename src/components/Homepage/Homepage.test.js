import React from 'react'
import { shallow } from 'enzyme'
import Homepage from './Homepage'

it('Should match the snapshot', () => {
  const wrapper = shallow(<Homepage />)
  expect(wrapper).toMatchSnapshot()
})