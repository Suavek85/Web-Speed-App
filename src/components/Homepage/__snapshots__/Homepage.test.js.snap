// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<section
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <Memo(SideNavigation) />
  <MainNavgation />
  <div
    className="block"
  >
    <Primary />
    <Side />
  </div>
</section>
`;
